hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/test/java/org/apache/hadoop/yarn/server/router/clientrm/TestFederationClientInterceptor.java

```diff
@@ -582,9 +583,9 @@ public void testGetClusterMetricsRequest() throws Exception {
     ClientMethod remoteMethod = new ClientMethod("getClusterMetrics",
         new Class[] {GetClusterMetricsRequest.class},
         new Object[] {GetClusterMetricsRequest.newInstance()});
-    Map<SubClusterId, GetClusterMetricsResponse> clusterMetrics = interceptor.
-        invokeConcurrent(new ArrayList<>(), remoteMethod, GetClusterMetricsResponse.class);
-    Assert.assertTrue(clusterMetrics.isEmpty());
+    Collection<GetClusterMetricsResponse> clusterMetrics = interceptor.
+        invokeConcurrent(remoteMethod, GetClusterMetricsResponse.class);
+    Assert.assertTrue(!clusterMetrics.isEmpty());
```

Comment:
Now the expectation is the opposite? What has changed?

Reply:
`invokeConcurrent`  This function originally contains 3 parameters

1. ArrayList<SubClusterId> clusterIds
2.ClientMethod request
3.Class<R> clazz

The first parameter is the SubClusters currently active.

In order to get the first parameter, all calling functions need to call the following function
`Map<SubClusterId, SubClusterInfo> subclusters = federationFacade.getSubClusters(true);`
I think we can put the fetching of the 1st parameter inside `invokeConcurrent`, which is more reasonable.

This unit test tests a situation where there is no Subcluster available, so an empty result is returned.
```
Map<SubClusterId, GetClusterMetricsResponse> clusterMetrics = interceptor.
        invokeConcurrent(new ArrayList<>(), remoteMethod, GetClusterMetricsResponse.class);
```

I removed the 1st parameter, it is a little difficult to simulate the situation where all Subclusters are unavailable.

I try to ensure that the original test logic can be completed, and I will refactor part of the code

Reply:
I will fix it.